apps/web/src/app/(pages)/(dashboard)/roles/page.tsx

```diff
@@ -22,6 +24,24 @@ export default function Roles({
     workEnvironment?: WorkEnvironmentType;
   };
 }) {
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
```

Comment:
We _could_ get away without using a `useEffect` or possibly a `useState` for tracking the errors!

We can define a `RolesSearchParam` schema in the `validators` package using Zod. Could be something like the following:

```typescript
const SearchParamsSchema = z.object({
  cycle: z.nativeEnum(WorkTerm).optional(),
  term: z.nativeEnum(WorkEnvironment).optional(),
});
```

We can share this object between the web app and the `list` endpoint for reviews. For validation, we can do something like the following:

```typescript
const validationResult = SearchParamsSchema.safeParse(searchParams);

const reviews = api.review.list.useQuery({
  options: validationResult.success ? validationResult.data : {},
});
```

At the same time, we can also use the same `success` property to display the error modal / toast:

```typescript
{!validationResult.success && (
        <ErrorAlert message="Invalid search parameters." />
      )}
```

If you want more specific error messages, you can modify the Zod schema like this:

```typescript
const SearchParamsSchema = z.object({
  cycle: z
    .nativeEnum(WorkTerm, {
      message: "Invalid cycle type",
    })
    .optional(),
  term: z
    .nativeEnum(WorkEnvironment, {
      message: "Invalid term type",
    })
    .optional(),
});
```

and get the errors using `validationResult.error`.